bpdm-gate/src/test/kotlin/org/eclipse/tractusx/bpdm/gate/controller/BusinessPartnerControllerIT.kt

```diff
@@ -80,8 +115,63 @@ class BusinessPartnerControllerIT @Autowired constructor(
         testHelpers.assertRecursively(searchResponsePage.content).isEqualTo(upsertResponses)
     }
 
+    @Test
+    fun `insert three business partners and check sharing state is pending and has taskid`() {
+        val upsertRequests = listOf(
+            BusinessPartnerNonVerboseValues.bpInputRequestFull,
+            BusinessPartnerNonVerboseValues.bpInputRequestMinimal,
+            BusinessPartnerNonVerboseValues.bpInputRequestChina
+        )
+        gateClient.businessParters.upsertBusinessPartnersInput(upsertRequests).body!!
+        val upsertExternalIds = listOf(
```

Comment:
I would make clear that you are using the same external ids.
One way would be to create tempary variables and use them consistently in the rest of the test.
```suggestion
        val externalId1 = BusinessPartnerNonVerboseValues.bpInputRequestFull.externalId
        val externalId2 = BusinessPartnerNonVerboseValues.bpInputRequestMinimal.externalId
        ...
        val externalIds = listOf(externalId1, externalId2, externalId3)
        ...
        val upsertSharingStatesRequests = listOf(
            ...
            externalId = externalId1
            ...
```